script.js

```diff
@@ -1,12 +1,95 @@
-//You can edit ALL of the code here
+
+const allEpisodes = getAllEpisodes();
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  
+  const oneEpisode = getOneEpisode();
+  allEpisodes.forEach(element => {makeDivForEpisode(element);
+  });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
-}
+// function makePageForEpisodes(episodeList) {
+//   const rootElem = document.getElementById("root");
+//   rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// }
 
 window.onload = setup;
+
+const episodesDiv = document.querySelector("#episodsHolder");
+
+// function for displaying all episodes
+
+function makeDivForEpisode (episode) {
+  const divContainer = document.createElement("div");
+  divContainer.className = "episodeContainer";
+
+  let title = document.createElement("h3");
+  let episodeCode = `S${(String(episode.season).padStart(2,'0'))}E${String(episode.number).padStart(2,'0')}`;
+  title.innerText = `${episode.name} - ${episodeCode}`;
+  divContainer.appendChild(title);
+
+  let avatar = document.createElement("img");
+  avatar.setAttribute("src", `${episode.image.medium}`);
+  divContainer.appendChild(avatar);
+
+  let description = document.createElement("p");
+  description.innerHTML = episode.summary;
+  divContainer.appendChild(description);
+
+  episodesDiv.appendChild(divContainer);
+
+}
+
+// search input
+document.querySelector("#search-input").addEventListener("input", searchText);
+
+function searchText(){
+  const searchInput = document.querySelector("#search-input").value.toLowerCase();
+  const filteredEpisodes = allEpisodes.filter(episode => {
+      if (episode.name.toLowerCase().includes(searchInput) || episode.summary.toLowerCase().includes(searchInput)){
```

Comment:
```suggestion
function searchText(value){
  const filteredEpisodes = allEpisodes.filter(episode => {
      if (episode.name.toLowerCase().includes(value) || episode.summary.toLowerCase().includes(value)){
```
So, diving logic this way makes it simple, reusable and readable.